src/hooks/useHashUrlBackwardsCompatibility.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+import { useHistory } from "react-router-dom";
+
+export function useHashUrlBackwardsCompatibility() {
+  const history = useHistory();
+
+  function onHashChange(event) {
+    const url = event.newURL.split("#").pop() ?? "/";
+    history.replace(url);
+  }
+
+  useEffect(() => {
+    window.addEventListener("hashchange", onHashChange);
+
+    return () => {
+      window.removeEventListener("hashchange", onHashChange);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (window.location.hash) {
```

Comment:
@calebjacob I'm curious, given the above event listener, why is this effect necessary?

Reply:
@charleslavon Good question! Unfortunately, the `hashchange` event listener doesn't fire on page load. So if you were to directly navigate to a URL with a hash (maybe from a bookmark), our 2nd effect is needed to remove the hash from the URL.